docs/source/why.rst

```diff
@@ -0,0 +1,217 @@
+Why Dask?
+=========
+
+This document gives high-level motivation on why people choose to adopt Dask.
+
+.. toctree::
+   :maxdepth: 1
+
+Python's role in Data Science
+-----------------------------
+
+Python has grown to become the dominant language both in data analytics, and
+general programming:
+
+.. image:: https://zgab33vy595fw5zq-zippykid.netdna-ssl.com/wp-content/uploads/2017/09/growth_major_languages-1-1024x878.png
+   :alt: Growth of major programming languages
+   :width: 75%
+
+This is fueled largely by libraries like Numpy, Pandas, and Scikit-Learn that
+are popular due to their combination of efficient algorithms and intuitive
+interfaces.
```

Comment:
Might also mention how Python is very interactive in a wide range of ways that are also useful for data analytics (e.g. Jupyter, asyncio, etc.).

Reply:
Resolved I think, by mentioning that Python also has libraries for plotting, interactive notebooks, etc..  I didn't mention libraries explicitly.  It would have been nice to mention Jupyter (which seems to have broad appeal) but it was hard to list that and not twenty plotting libraries.

Reply:
Sure. Though the gist of my suggestion was different than say listing a bunch of plotting libraries.

Was trying to say Python is a strong language because it has strong analytic libraries and strong web friendly interactive tools, which makes it ideal for doing analysis in a variety of different configuration. Of course that can be nicely followed up by Dask leverages these two strengths and joins them into a powerful, interactive analytic tool. ;)

Does that make more sense? Of course feel free to not include that if you would rather go in a different direction.

Reply:
Right, I understand and I agree.  I was trying to communicate this and ended up with the following:

```
+This is fueled both by computational libraries like Numpy, Pandas, and
+Scikit-Learn and by a wealth of libraries for visualization, interactive
+notebooks, collaboration, and so forth.
```

In my comment above I was trying to apologize for not mentioning Jupyter/Tornado/asyncio in particular, and instead switching to general terms.  I tried doing this at first to copy the structure set up in the first part of that sentence, which lists popular and well known libraries, but I wasn't able to.  Mostly this was because there wasn't a clear set of libraries to select.  I instead opted for general buckets.  I hope that this is ok.  

I also apologize for merging right away, happy to edit after-the-fact if desired.

Reply:
No worries. It's not crucial.

Just was thinking of the justification (advertisement) for Python here in terms of exactly what Dask/Distributed provides IOW numerics/async. Thus one justification could effectively be used for why Python is a good language and lead right into why it is a good fit for the problem being solved here. The libraries are certainly important as well and it's reasonable to mention them too.

As I've not been doing as much reading and writing of docs, it's possible I'm totally off base. Not saying we need to use the argument above either. Happy to defer to your judgement if you think the current content is a better fit.